ENT-11569: Added documentation for mailto, mailfrom, smtpserver and maxmaillines (#2880)
cfengine/masterfiles: MPF.md

### mailto

The address that `cf-execd` should email agent output to. Defaults to `root@$(default:def.domain)`.

This setting can be customized via Augments, for example:

```json
{
  "variables": {
    "default:def.mailto": {
        "value": "cfengine-maintainers@example.com",

>         "comment": "Email sent from cf-execd when output differs from the prior execution should be delievered to this distribution group for review"

Reviewer: ```suggestion
        "comment": "When output differs from the prior execution cf-execd will deliver ??? to this Email address for review"
```

What is sent exactly? the diff, the new output, the previous output, both?

Author: It's the standard  cf-execd email behavior.

If the output differs from the previous run, all output is emailed (up to maxmaillines). So, e.g. if you put a report with $(sys.date) in your policy, your going to get all the output from the agent run each time.

Assuming no other output change, if you report "Hello world" you get an email with all output and no further email until the output changes.

    }
  }
}
```

### mailfrom

The address that output mailed from `cf-execd` should come from. Defaults to `root@$(sys.uqhost).$(def.domain)`.

This setting can be customized via Augments, for example:

```json
{
  "variables": {
    "default:def.mailfrom": {
        "value": "cfengine@example.com",
        "comment": "Email sent from cf-execd should come from this address"
    }
  }
}
```

### smtpserver

The SMTP server that `cf-execd` should use to send emails. Defaults to `localhost`.

This setting can be customized via Augments, for example:

```json
{
  "variables": {
    "default:def.smtpserver": {

>         "value": "cfengine@example.com",

Reviewer: ```suggestion
        "value": "smtp.example.com",
```

And is this assuming a plain SMTP endpoint not encrypted or requiring user information for authentication?

Author: yep.

        "comment": "The smtp server that should be used when sending email from cf-execd."
    }
  }
}
```

### mailmaxlines

> The maximum number of lines of output that `cf-execd` will email. Defaults to `30`.

Reviewer: This is in relation to the difference between previous execution mentioned earlier? Might want to mention this is line of "execution difference report".
